src/components/autocompleter/autocompleter.test.jsx

```diff
@@ -0,0 +1,139 @@
+    });
+  });
+
+  describe('errorText', () => {
+    it('sets the input to be invalid', () => {
+      render(<Autocompleter {...requiredProps} errorText="not valid" />);
+      expect(screen.getByLabelText('Test label')).not.toBeValid();
+    });
+  });
+
+  describe('id API', () => {
+    it('accepts an ID', () => {
+      render(<Autocompleter {...requiredProps} id="this-is-an-id" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('id', 'this-is-an-id');
+    });
+  });
+
+  describe('label API', () => {
+    it('accepts a label', () => {
+      render(<Autocompleter {...requiredProps} label="Bar" />);
+      expect(screen.getByLabelText('Bar')).toBeDefined();
+    });
+  });
+
+  describe('placeholder API', () => {
+    it('accepts a placeholder', () => {
+      render(<Autocompleter {...requiredProps} placeholder="I am place" />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('placeholder', 'I am place');
+    });
+  });
+
+  describe('readOnly API', () => {
+    it('sets the readonly attribute', () => {
+      render(<Autocompleter {...requiredProps} readOnly />);
+      expect(screen.getByLabelText('Test label')).toHaveAttribute('readonly', '');
+    });
+
+    it('unsets the readonly attribute', () => {
+      render(<Autocompleter {...requiredProps} readOnly={false} />);
+      expect(screen.getByLabelText('Test label')).not.toHaveAttribute('readonly', '');
+    });
+  });
+
+  describe('required API', () => {
+    it('sets the required attribute', () => {
+      render(<Autocompleter {...requiredProps} required />);
+      expect(screen.getByLabelText('Test label')).toBeRequired();
+    });
+
+    it('unsets the required attribute', () => {
+      render(<Autocompleter {...requiredProps} required={false} />);
+      expect(screen.getByLabelText('Test label')).not.toBeRequired();
+    });
+  });
+
+  describe('value API', () => {
+    it('accepts a value', async () => {
+      const value = { id: 22, name: 'cool dude' };
+      render(<Autocompleter {...requiredProps} value={value} />);
+
+      expect(screen.getByLabelText('Test label')).toHaveValue('cool dude');
+    });
+  });
+
+  describe('models API', () => {
+    it('accepts a an array of models', async () => {
+      const models = [{ id: 22, name: 'cool dude' }, { id: 33, name: 'neato burrito' }];
+      render(<Autocompleter {...requiredProps} models={models} />);
+
+      userEvent.click(screen.getByLabelText('Test label'));
+      expect(screen.queryByText('cool dude')).toBeInTheDocument();
+      expect(screen.queryByText('neato burrito')).toBeInTheDocument();
+    });
+  });
+
+  describe('forwardRef API', () => {
+    it('can be used to get value as array of selected id', async () => {
+      const ref = createRef();
+      render(<Autocompleter {...requiredProps} value={{ name: 'neat', id: 11 }} ref={ref} />);
+
+      expect(ref.current.value).toStrictEqual({ name: 'neat', id: 11 });
+    });
+  });
```

Comment:
I really like this spec.

It makes it super clear _why_ using the ref is useful, and I think refs are opaque and odd enough that not everyone in the know would necessarily get it, so this spec is helpful